src/intrinsics/ecma262/Object.js

```diff
@@ -152,7 +152,7 @@ function applyObjectAssignSource(
 
 function tryAndApplySourceOrRecover(
   realm: Realm,
-  nextSource: ObjectValue | AbstractObjectValue,
+  nextSource: Value,
```

Comment:
Is this type weakening really needed?

Reply:
Yeah, this only happened recently after updating Flow I think but no one changed the file. Strange one but nextSource might be `null` or `undefined`.